### ISSUE TITLE
24

### ISSUE THREAD

**Issue by halj3317-bit** [halj3317-bit/2#13]
import pandas as pd
from sklearn.feature_selection import VarianceThreshold
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import r2_score

df = pd.read_csv("/mnt/data/data_day24_selection.csv")

X = df.drop("target", axis=1)
y = df["target"]

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=1)

model1 = LinearRegression()
model1.fit(X_train, y_train)
pred1 = model1.predict(X_test)
print("قبل:", r2_score(y_test, pred1))

selector = VarianceThreshold(0.01)
X2 = selector.fit_transform(X)
X2 = pd.DataFrame(X2)

corr = X2.corr().abs()
drop_cols = []

for i in range(len(corr.columns)):
    for j in range(i):
        if corr.iloc[i, j] > 0.9:
            drop_cols.append(corr.columns[i])

X3 = X2.drop(columns=drop_cols)

X_train2, X_test2, y_train2, y_test2 = train_test_split(X3, y, test_size=0.2, random_state=1)

model2 = LinearRegression()
model2.fit(X_train2, y_train2)
pred2 = model2.predict(X_test2)
print("بعد:", r2_score(y_test2, pred2))
